### PR TITLE
implement IsRandEquivalent

### PR DESCRIPTION
**Pull request by qindazhu**
Implement `IsRandEquivalent` with some tests. Also fixed some style issues caused by @danpovey 's last PR. Please merge this PR first, otherwise you need to fix those style issues again to pass style check.

### CODE REVIEW

**Comment by danpovey** [comment]
Looks fine to me.
You guys can merge once you think it's OK.

File: k2/csrc/determinize.cc [hunk 1]
```diff
@@ -68,15 +66,18 @@ struct DetStateElement {
...
+  // `base_state`.
+  // The sequence of symbols can be found by tracing back one of the
+  // DetStateElements
+  // in the doubly linked list (it doesn't matter which you pick, the result
+  // will be the
   // same.
   int32_t seq_len;
 
-  bool normalized { false };
+  bool normalized{false};
```
Comment (csukuangfj) [inline k2/csrc/determinize.cc:78]: For consistency, other members should also be initialized.

File: k2/csrc/fsa_equivalent.cc [hunk 2]
```diff
@@ -14,9 +14,54 @@
...
 
+bool IsRandEquivalent(const Fsa &a, const Fsa &b, std::size_t npath /*=100*/) {
+  // We will do `intersect` later which requires either `a` or `b` is
+  // epsilon-free, considering they should hold same set of arc labels, so both
+  // of them should be epsilon-free.
+  // TODO(haowen): call `RmEpsilon` here instead of checking.
+  if (!IsEpsilonFree(a) || !IsEpsilonFree(b)) return false;
+
+  Fsa connected_a, connected_b, valid_a, valid_b;
+  Connect(a, &connected_a);
```
Comment (csukuangfj) [inline k2/csrc/fsa_equivalent.cc:29]: You need to check the return status of `Connect`.

File: k2/csrc/determinize.cc (hunk 1 above)
Comment (qindazhu) [reply k2/csrc/determinize.cc:78]: Please donot comments on this file as it it even not compiled. I just fix style issues to pass the style check.

File: k2/csrc/fsa_equivalent.h [hunk 3]
```diff
@@ -15,10 +16,11 @@
...
 
 /*
-  Returns true if the Fsa `a` is equivalent to `b`.
-  CAUTION: this one will be quite hard to implement.
+  Returns true if the Fsa `a` is stochastically equivalent to `b` by randomly
+  generating `npath` paths from one of them and then checking if the
+  paths exist in the other one.
  */
-bool IsEquivalent(const Fsa &a, const Fsa &b);
+bool IsRandEquivalent(const Fsa &a, const Fsa &b, std::size_t npath = 100);
```
Comment (csukuangfj) [inline k2/csrc/fsa_equivalent.h:23]: Why is `npath` of type `std::size_t`?

File: k2/csrc/fsa_equivalent.cc (hunk 2 above)
Comment (qindazhu) [reply k2/csrc/fsa_equivalent.cc:29]: I dont's care the return value (is_acyclic) here. Even if it failed, leave `connected_a` with empty is fine here

File: k2/csrc/fsa_equivalent.h (hunk 3 above)
Comment (qindazhu) [reply k2/csrc/fsa_equivalent.h:23]: as it is the number of paths we will generate, a positive value

**Comment by qindazhu** [comment]
Please merge this if there's any other issues.

**Pull request merged**